### ISSUE TITLE
Phantom pet eggs

### ISSUE THREAD

**Issue by shanaqui** [HabitRPG/habitica-android#1460]
Users have been reporting this on and off for a while.

> "Hi, so I've had this glitch in my egg inventory for probably at least 9-12 months.
> 
> I think I used to know what the 12 missing items were, but I forgot ages ago.
> 
> If I tap on them, the pop up for selling an egg shows up and offers me 'null' gold. I haven't tried to sell any for null gold, at least not in the last few months or so.
> 
> It's not a huge problem, especially since I can't remember what they were anymore, but it is odd and it is a glitch and it hasn't ever gone away that I've caught."
> 
> Device: samsung SM-A515U
> Android Version: 29
> AppVersion: Version 3.1.1 (2690)
> User ID: 31ff377b-690f-4328-952e-0aab8fb8e80e
> Level: 134
> Class: wizard
> Is in Inn: false
> Uses Costume: true
> Custom Day Start: 3
> Timezone Offset: 300

They don't have twelve of anything in their inventory.

![Screenshot_20210114-020745_Habitica](https://user-images.githubusercontent.com/8144640/104735016-fb401c80-5738-11eb-804e-7c473fcccb55.jpg)

**Comment by saraolson** [comment]
I also have this problem, and have brought it up but we've never found a cause or a fix for them. They dont seem to be actual items, just some bad data? ie nothing is missing that the account should have access to

**Comment by shanaqui** [comment]
Also currently being experienced by this user:

Device: samsung Galaxy S8
Android Version: 28
AppVersion: Version 3.1.2 (2694)
User ID: 243484ac-bb24-4170-8fe7-bf3a53e0684b
Level: 41
Class: wizard
Is in Inn: false
Uses Costume: false
Custom Day Start: 7
Timezone Offset: 0

**Comment by shanaqui** [comment]
And with an interesting new detail:

> Device: samsung SM-J600FN
> Android Version: 26
> AppVersion: Version 3.2 (2832)
> User ID: b24c2422-7212-448f-a5bb-c9603d25c861
> Level: 7
> Class: warrior
> Is in Inn: false
> Uses Costume: false
> Custom Day Start: 0
> Timezone Offset: 0
> Details:
> 
>  Please see screenshot for the inventory_present item in my Egg Inventory. It only gives the option to Sell for null Gold. 

Maybe something weird with the subscriber giftbox?

![Screenshot_20210305-211621_Habitica](https://user-images.githubusercontent.com/8144640/110248044-196f0f80-7f67-11eb-84dc-985f4c5d2e66.jpg)

**Comment by cvuorinen** [comment]
I looked into this as I was also seeing that "inventory_present" thing in my eggs list. Managed to figure out that it's actually the mystery items that mistakenly had `itemType = "eggs"` applied to them in certain conditions.

Not sure if the issue that's in the first screenshot is different, feels like probably it is not the same as it does not show the text "inventory_present" and also it's not likely that any user has 12 unopened mystery items.

**Comment by saraolson** [comment]
@cvuorinen i dont think they need to be unopened. i have a test account with inv_present at 62 and 62 unopened mystery items. when i opened the items, the inventory present stayed at 62. 🤔

**Comment by saraolson** [comment]
i see the initial report opened 8 mystery items jan 10 2021, and 4 in march 2020 🤔

**Comment by cvuorinen** [comment]
> @cvuorinen i dont think they need to be unopened. i have a test account with inv_present at 62 and 62 unopened mystery items. when i opened the items, the inventory present stayed at 62. 🤔 

Yeah, might be that the list just didn't update yet. I found it notoriously hard to get any actual changes I made in the code to get updated to that list. Just pulling down does not update it, going to settings page didn't update it, only way I found to be able to test the changes I made was to complete logout, and then log back in.

**Comment by saraolson** [comment]
oh yep that did the trick down to 60 n 60 now
